[翻译完毕] The life of a Taichi kernel(compilation.po)

isdanni opened the pull request:
Not sure about Taichi AST `lowering pass`, I translated to `降低通行`, please correct me if I'm wrong;

Issue: #28

archibate commented on locales/zh_CN/LC_MESSAGES/compilation.po:
```diff
@@ -81,150 +90,172 @@ msgid ""
+"applied to the Python AST so that the Taichi frontend can recognize it."
 msgstr ""
+"当发生新的实例化时，Taichi前端编译器会将内核AST转换为Python脚本，该脚本在"
+"执行时会发出Taichi前端AST。 基本上，一些补丁已应用于Python AST，以便"
```
```suggestion
"执行时会产生Taichi前端AST。 主要是在Python AST上应用一些补丁，以便"
```

archibate commented on locales/zh_CN/LC_MESSAGES/compilation.po:
```diff
@@ -81,150 +90,172 @@ msgid ""
 #: ../../compilation.rst:97
 msgid "Access lowering"
-msgstr ""
+msgstr "降低访问权限"
```
```suggestion
msgstr "低阶化访问"
```
这里的lowering指的是从前端到后端的过程。

archibate commented on locales/zh_CN/LC_MESSAGES/compilation.po:
```diff
@@ -81,150 +90,172 @@ msgid ""
 "Finally, the optimized SSA IR is fed into the LLVM IR codegen, and LLVM "
 "JIT generates high-performance executable CPU/GPU programs."
 msgstr ""
+"最后，经过优化的SSA IR 被反馈进 LLVM IR 码元，并且LLVM JIT生成了高性能可"
```
```suggestion
"最后，经过优化的SSA IR 被送进 LLVM IR 代码生成器，并且由LLVM JIT生成高性能的可"
```

archibate reviewed (commented):
Btw, I think we should translate or mention that AST is 抽象语法树, wdyt?

isdanni committed:
update compilation

isdanni commented:
> Btw, I think we should translate or mention that AST is 抽象语法树, wdyt?

Agree :thumbsup: Updated with this commit [5386bfb](https://github.com/taichi-dev/taichi-docs-zh-cn/commit/5386bfb9f77b1c57b03f11c7e188f78ebfeccc6c), please help check

archibate commented on locales/zh_CN/LC_MESSAGES/compilation.po:
```diff
@@ -3,58 +3,63 @@
 #: ../../compilation.rst:4
 msgid "The life of a Taichi kernel"
-msgstr ""
+msgstr "Taichi内核的生命"
```
```suggestion
msgstr "Taichi内核的生命周期"
```

archibate commented on locales/zh_CN/LC_MESSAGES/compilation.po:
```diff
@@ -81,150 +92,173 @@ msgid ""
 "hierarchical static single assignment (SSA) IR, which allows a series of "
 "further IR passes to happen, such as"
 msgstr ""
+"Taichi 抽象语法树降低通行Taichi前端IR转换为分层静态单任务(SSA) IR，从而允"
+"许进行进一步一系列的IR通行，例如"
```
IMO `pass` is a noun term that stands for the *IR transformer*. Let's say `pass` -> `过程`.
```suggestion
"AST 的降阶过程 (lowering pass) 会将前端 IR 转换为分层静态单任务 (SSA) 的 IR，"
"从而可以用更多的过程进一步处理 IR，例如"
```

archibate reviewed (commented):
Thank for the patient! Btw, we may also mention that IR is 中间表示代码 where it first occurred?

isdanni committed:
update IR & SSA

isdanni commented:
> Thank for the patient! Btw, we may also mention that IR is 中间表示代码 where it first occurred?

Yes agree :+1:  Just added an update [here](https://github.com/taichi-dev/taichi-docs-zh-cn/pull/29/commits/f953caddb1eb99673d592004357d83c662933312)

Also, I added a ref link for SSA(Static Single Assignment)

isdanni committed:
update 中间码 to 中间表示码

archibate reviewed (approved):
Cool! Everything looks great now!
